source/agora/consensus/data/Block.d

```diff
@@ -796,7 +796,7 @@ version (unittest)
             // rc = r used in signing the commitment
             const Scalar rc = Scalar(hashMulti(key.secret, "consensus.signature.noise",
                 cycleForValidator(key.address)));
-            const Scalar r = rc + challenge; // make it unique each challenge
+            const Scalar r = rc + Scalar(block.header.height.hashFull());
```

Comment:
Why do we hash the height ?? It should be the pre-image.

Reply:
Using the height hashed is a quick win solution (from using the block hash) to prevent nodes signing more than one block at a certain height. Can you please explain why using the `pre-image` is preferred here?

Reply:
After discussion we decided to use pre-image to be more secure as an attack might still be possible as the hashed height is predictable for all blocks whereas the pre-image are only revealed for several blocks ahead at most.